### ISSUE TITLE
False positive `arguments-differ` with defaults

### ISSUE THREAD

**Issue by socketpair** [PyCQA/pylint#5338]
### Bug description

```python
class Base:
    def method(self, arg: str):
        print(arg)


class Derived(Base):
    def method(self, arg: str, param1: int = 42, *, param2: int = 42):
        print(arg, param1, param2)


xxx: Base = Derived()
xxx.method('blah')
```

### Pylint output

```shell
 W0221: Number of parameters was 2 in 'Base.method' and is now 4 in overridden 'Derived.method' method (arguments-differ)
```

### Expected behavior

no arguments-differ error in the case.

### Pylint version

```shell
pylint 2.9.6
astroid 2.6.6
Python 3.9.7 (default, Aug 30 2021, 00:00:00) 
[GCC 11.2.1 20210728 (Red Hat 11.2.1-1)]
```

### OS / Environment

Fedora 34

**Comment by socketpair** [comment]
This is a natural case for extending base classes. So every code expecting base class CAN call the new (overridden) function without any problems.

**Comment by Pierre-Sassoulas** [comment]
``arguments-differ`` is a check of the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle) that this example violate. You could not replace a call to ``Derived`` by ``Base`` because there are added argument. If you want to violate the LSP you should disable ``arguments-differ``.

**Comment by socketpair** [comment]
No, please give proofs where this case violates LSP. I strongly consider it's not.

**Comment by Pierre-Sassoulas** [comment]
It appears you're right and that the LSP do not say anything about the base class being able to be a substitute for its subtype only the other way around. So ``argument-differ`` seems to be stricter than the LSP. We already introduced a watered down version with ``argument-renamed`` in https://github.com/PyCQA/pylint/pull/4467, so it would make sense to create a LSP compatible message.

**Comment by DanielNoord** [comment]
I believe this is a duplicate of #1556
